src/components/Menu/MenuItem/MenuItem.tsx

```diff
@@ -199,12 +208,25 @@ const MenuItem: VibeComponent<MenuItemProps> & {
       [setSubMenuIsOpenByIndex, index, closeMenu]
     );
 
-    const mergedRef = useMergeRefs({ refs: [ref, referenceElementRef] });
-
     const renderSubMenuIconIfNeeded = () => {
       if (!hasChildren) return null;
 
-      return (
+      return splitButton ? (
+        <div className={styles.subMenuIconWrapper}>
+          <Divider direction={DirectionType.VERTICAL} className={styles.divider} />
+          <IconButton
+            icon={DropdownChevronRight}
+            className={styles.subMenuSplitButtonIcon}
+            kind={IconButton.kinds.TERTIARY}
+            size={null}
+            iconClassName={styles.iconButton}
+            onClick={() => {}}
```

Comment:
Why it's needed?

Reply:
because it created unwanted behavior.
From what i understood- we did not want to support an on click on the icon button.
![ezgif com-video-to-gif (14)](https://github.com/mondaycom/monday-ui-react-core/assets/139220157/50bfafe4-8e4d-42df-90df-9838851c978d)

Reply:
When clicking on `IconButton` it's causing MenuItem onClick to prevent it you need to check onClick target and compare it with refs in `useMenuItemKeyboardEvents`